From c7c0000a4e95fe259e89c644cc19e4d37565bc7c Mon Sep 17 00:00:00 2001
From: Rufus Hamade <rufus.hamade@imgtec.com>
Date: Mon, 29 Feb 2016 15:14:51 +0000
Subject: [PATCH 102/102] pvr_dri: remove bpp parameter from
 PVRDRIBufferCreateFrom(Fd|Name)

This wasn't used for anything useful and the wrong value was being
provided in some cases. Go ahead and remove it.
---
 src/mesa/drivers/dri/pvr/pvr_dri_support.h | 6 ++----
 src/mesa/drivers/dri/pvr/pvrdrawable.c     | 3 +--
 src/mesa/drivers/dri/pvr/pvrimage.c        | 6 ++----
 3 files changed, 5 insertions(+), 10 deletions(-)

diff --git a/src/mesa/drivers/dri/pvr/pvr_dri_support.h b/src/mesa/drivers/dri/pvr/pvr_dri_support.h
index 026c3bc..2704df9 100644
--- a/src/mesa/drivers/dri/pvr/pvr_dri_support.h
+++ b/src/mesa/drivers/dri/pvr/pvr_dri_support.h
@@ -250,16 +250,14 @@ PVRDRIBufferImpl *PVRDRIBufferCreateFromName(PVRDRIScreenImpl *psScreenImpl,
 					     int iWidth,
 					     int iHeight,
 					     unsigned int uiStride,
-					     unsigned int uiOffset,
-					     unsigned int uiBpp);
+					     unsigned int uiOffset);
 
 PVRDRIBufferImpl *PVRDRIBufferCreateFromFd(PVRDRIScreenImpl *psScreenImpl,
 					   int iFd,
 					   int iWidth,
 					   int iHeight,
 					   unsigned int uiStride,
-					   unsigned int uiOffset,
-					   unsigned int uiBpp);
+					   unsigned int uiOffset);
 
 void PVRDRIBufferDestroy(PVRDRIBufferImpl *psBuffer);
 
diff --git a/src/mesa/drivers/dri/pvr/pvrdrawable.c b/src/mesa/drivers/dri/pvr/pvrdrawable.c
index cbbca11..599d00d 100644
--- a/src/mesa/drivers/dri/pvr/pvrdrawable.c
+++ b/src/mesa/drivers/dri/pvr/pvrdrawable.c
@@ -474,8 +474,7 @@ static void *PVRDRI2ObjectCacheInsert(void *pvCreateData, void *pvInsertData)
 					   psDRIDrawable->w,
 					   psDRIDrawable->h,
 					   psDRIBuffer->pitch,
-					   0,
-					   psPVRDrawable->ePixelFormat);
+					   0);
 	if (!psPVRBuffer->uBacking.sDRI2.psBuffer)
 	{
 		free(psPVRBuffer);																					
diff --git a/src/mesa/drivers/dri/pvr/pvrimage.c b/src/mesa/drivers/dri/pvr/pvrimage.c
index 43f5044..f912cc4 100644
--- a/src/mesa/drivers/dri/pvr/pvrimage.c
+++ b/src/mesa/drivers/dri/pvr/pvrimage.c
@@ -208,8 +208,7 @@ CreateImageSharedFromNames(__DRIscreen *screen,
 						   width >> psFormat->sPlanes[i].uiWidthShift,
 						   height >> psFormat->sPlanes[i].uiHeightShift,
 						   strides[i],
-						   offsets[i],
-						   PVRDRIPixFmtGetBPP(psFormat->sPlanes[i].eIMGPixelFormat));
+						   offsets[i]);
 		if (!shared->apsBuffers[i])
 		{
 			errorMessage("%s: Failed to create plane %d for shared image\n", __func__, i);
@@ -296,8 +295,7 @@ CreateImageSharedFromDmaBufs(__DRIscreen *screen,
 						 width >> psFormat->sPlanes[i].uiWidthShift,
 						 height >> psFormat->sPlanes[i].uiHeightShift,
 						 strides[i],
-						 offsets[i],
-						 PVRDRIPixFmtGetBPP(psFormat->sPlanes[i].eIMGPixelFormat));
+						 offsets[i]);
 		if (!shared->apsBuffers[i])
 		{
 			errorMessage("%s: Failed to create plane %d for shared image\n", __func__, i);
-- 
1.9.1

